wpimath/src/main/java/edu/wpi/first/math/controller/SimpleMotorFeedforward.java

```diff
@@ -39,62 +56,55 @@ public SimpleMotorFeedforward(double ks, double kv, double ka) {
     if (ka < 0.0) {
       throw new IllegalArgumentException("ka must be a non-negative number, got " + ka + "!");
     }
+    if (periodSeconds <= 0.0){
+      throw new IllegalArgumentException("period must be a positive number, got " + periodSeconds + "!");
+    }
+    this.plant = LinearSystemId.identifyVelocitySystem(this.kv, this.ka);
+    this.feedforward = new LinearPlantInversionFeedforward<>(plant, periodSeconds);
+
+    this.r = MatBuilder.fill(Nat.N1(), Nat.N1(), 0.0);
+    this.nextR = MatBuilder.fill(Nat.N1(), Nat.N1(), 0.0);
   }
 
   /**
-   * Creates a new SimpleMotorFeedforward with the specified gains. Acceleration gain is defaulted
-   * to zero. Units of the gain values will dictate units of the computed feedforward.
+   * Creates a new SimpleMotorFeedforward with the specified gains and period. The period is defaulted to 20 ms.
+   * Units of the gain values will dictate units of the computed feedforward.
    *
    * @param ks The static gain.
    * @param kv The velocity gain.
+   * @param ka The acceleration gain.
+   * @throws IllegalArgumentException for kv &lt; zero.
+   * @throws IllegalArgumentException for ka &lt; zero.
    */
-  public SimpleMotorFeedforward(double ks, double kv) {
-    this(ks, kv, 0);
+  public SimpleMotorFeedforward(double ks, double kv, double ka) {
+    this(ks, kv, ka, 0.020);
   }
 
   /**
-   * Calculates the feedforward from the gains and setpoints.
+   * Creates a new SimpleMotorFeedforward with the specified gains. Acceleration gain is defaulted
+   * to zero. The period is defaulted to 20 ms.  Units of the gain values will dictate units of the computed feedforward.
    *
-   * @param velocity The velocity setpoint.
-   * @param acceleration The acceleration setpoint.
-   * @return The computed feedforward.
+   * @param ks The static gain.
+   * @param kv The velocity gain.
    */
-  public double calculate(double velocity, double acceleration) {
-    return ks * Math.signum(velocity) + kv * velocity + ka * acceleration;
+  public SimpleMotorFeedforward(double ks, double kv) {
+    this(ks, kv, 0, 0.020);
   }
 
   /**
-   * Calculates the feedforward from the gains and setpoints.
+   * Calculates the feedforward from the gains and setpoints assuming discrete control.
    *
    * @param currentVelocity The current velocity setpoint.
    * @param nextVelocity The next velocity setpoint.
-   * @param dtSeconds Time between velocity setpoints in seconds.
    * @return The computed feedforward.
    */
-  public double calculate(double currentVelocity, double nextVelocity, double dtSeconds) {
-    var plant = LinearSystemId.identifyVelocitySystem(this.kv, this.ka);
-    var feedforward = new LinearPlantInversionFeedforward<>(plant, dtSeconds);
-
-    var r = MatBuilder.fill(Nat.N1(), Nat.N1(), currentVelocity);
-    var nextR = MatBuilder.fill(Nat.N1(), Nat.N1(), nextVelocity);
+  public double calculate(double currentVelocity, double nextVelocity) {
```

Comment:
I like the idea of the feedforward classes being discrete-only, but I'm not sure how to make the migration non-breaking in Java because the two calculate() functions have the same signature.

Reply:
could we have one method called calculate and one called calculateDiscrete (or something similar) so as not to be a breaking change at least at this moment?

Reply:
The problem with that is it takes 4 years to migrate the API in a non-breaking way: year 1 adds calculateDiscrete() and deprecates calculate(), year 2 removes calculate(), year 3 adds calculate() with the new behavior and deprecates calculateDiscrete(), and year 4 removes calculateDiscrete(). We don't give calculate() new behavior in year 2 to avoid confusion.

Reply:
Okay.  Just spitballing here.  Separate class?  SimpleFeedforwardDiscrete

Reply:
or should I just wait on this one for discussion at the top level.

Reply:
Could we use the Java units class for the signature type of the other method?

Reply:
We could potentially use units, yea.

Reply:
Okay.  I'll start working on the modification to Java.  Again I'll wait for feedback before working on the C++ version.  
To be clear I will leave the calculate method with double velocity and double acceleration as is and add one with velocity measures for current and next velocity.  The calculate method with 3 doubles (current, next velocities and time will still be excluded right?

Reply:
Correction:  the calcluate method with doubles would be velocity and acceleration and use the continuous case.

Reply:
Okay.  I'll start working on a modification to this PR.  
So ideally

Reply:
I'd like to see the velocity-acceleration overload be deprecated.

Reply:
And what year 2025, 6?

Reply:
Okay I've added the changes.  Is this what you think might work better?

Reply:
@calcmogul 
Okay so I've added the changes as I've interpreted them.  This required lots of modifications to examples and tests.  

In fact I know a test is failing because ka cannot be zero in identifyVelocitySystem, but the test/example is setting it to 0 in the SimpleFeedforward constructor.

Reply:
I need some more guidance before I work on the C++ version

Reply:
If Ka is close to zero, you can fall back to just the Ks and Kv feedforward. The issue is when Ka approaches zero, A = -∞ and B = ∞. It resolves to a system with a velocity state and input.

Reply:
So use the continuous version of the feedforward in that case?

Reply:
Yes, because when Ka = 0, the continuous and discrete feedforwards are the same.

Reply:
Okay.  updates made to constructor and calculate method.  
I'm failing tests at line 53 of exponentialProfileTest.

Reply:
The dt in the exponential profile test doesn't match the default dt of SimpleMotorFeedforward. You should update the SimpleMotorFeedforward creation to pass the test's dt.

Reply:
Ah.  Thank you.  Little things right.  
updates pushed.  I'll start working on the C++ version.  Anything on the Java version that you would recommend modifying?

Reply:
How do I mark a method as deprecated in C++.

Reply:
Copy-paste the `@deprecated` doc tag from Java, then use the attribute `[[deprecated("message here")]]`.